### PR TITLE
python_requirements_info: don't overwrite results in 'mismatched' dict key

### PR DESCRIPTION
**Pull request by wbh1**
##### SUMMARY

Whichever mismatched package is evaluated last is the value stored in the
'mismatched' key. Instead, it should have a subdict for each pkg that is mismatched
to keep in line with its documented usage and how the `valid` return value is handled.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

python_requirements_info

##### ADDITIONAL INFORMATION

With `dependencies: ['salt==3003.3', 'jinja2<1']`:

Before:
```json
salt-master-us-east | SUCCESS => {
    "mismatched": {
        "desired": "jinja2>3",
        "installed": "2.10"
    },
}
```

After:
```json
salt-master-us-east | SUCCESS => {
    "mismatched": {
        "jinja2": {
            "desired": "jinja2>3",
            "installed": "2.10"
        },
        "salt": {
            "desired": "salt==3003.3",
            "installed": "3003"
        }
    },
}
```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)? Thanks!

**Comment by wbh1** [comment]
@felixfontein Done. Thank you!

File: changelogs/fragments/4078-python_requirements_info.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - python_requirements_info - store ``mismatched`` return values per package
```
Comment (felixfontein) [inline changelogs/fragments/4078-python_requirements_info.yaml:2]:
```suggestion
  - python_requirements_info - store ``mismatched`` return values per package as documented in the module (https://github.com/ansible-collections/community.general/pull/4078).
```

**Review by felixfontein — APPROVED** [review]
Looks good! If nobody complains, I'll merge this by the end of this week.

**Comment by felixfontein** [comment]
@wbh1 thanks for your contribution!

**Pull request merged**